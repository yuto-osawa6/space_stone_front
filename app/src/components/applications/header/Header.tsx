import { useEffect, useRef, useState } from "react";
import { BsFillSuitHeartFill, BsListUl } from "react-icons/bs"
import { MdDriveFileMoveOutline, MdHome, MdOutlineArticle, MdRateReview, MdSearch } from "react-icons/md";
import { useDispatch } from "react-redux";
import { useScrollPosition } from '@n8tb1t/use-scroll-position';
import { HiOutlineSearchCircle } from "react-icons/hi";
import { IoMdMenu } from "react-icons/io";
import Link from 'next/link'
import { AiOutlineComment, AiOutlineHome } from "react-icons/ai"
import { RiArticleLine } from "react-icons/ri"
import { UserCertification } from "../user/UserCertification";
import { SubMenuAction } from "@/store/submenu/actions";
import { useLocale } from "@/lib/ini/local/local";
import { useWindowDimensions } from "@/hook/useWindowResize";
import { url } from "@/utils/config";
import { useScroll } from "@/hook/useScroll";
import { useRouter } from "next/router";
import { useExecLeft } from "@/lib/api/left";
import LeftStyle from "../left/leftMenus/LeftStyles";
import LeftGenre from "../left/leftMenus/LeftGenres";
import { LeftsArticles } from "@/components/applications/left/leftMenus/LeftArticles";
import { LeftsReviews } from "../left/leftMenus/LeftReviews";
import { LeftsThreads } from "../left/leftMenus/LeftsThreads";


type Props = {
  locationNumber: number | undefined
}
export const Header:React.FC<Props> = function HeaderFunc(Props){
  const a = useScroll()
  // console.log(a)
  // useScrollPosition
  const dispatch = useDispatch()
  const [showMenu, setShowMenu] = useState<boolean>(true);
  const [currentPostion,setCurrentPositon] = useState<number>(0)
  // useScroll(({scrollY,preveScroll}) => {
  //   // if (scrollY>-1){
  //   //   setShowMenu(true)
  //   //   return
  //   // }
  //   // const visible = scrollY > preveScroll;
  //   // setShowMenu(visible);
  // },[])

  useEffect(()=>{
    // console.log("aaa")
    // console.log(a.scrollY ,a.prevScrollY )
    // console.log(showMenu)
    // console.log(a.scrollY<1)
    // console.log("aaa")
    // console.log(a.scrollY > a.prevScrollY)
    if (a.scrollY<56){
      console.log("aaa34")
      setShowMenu(true)
      return
    }
    const visible = a.scrollY < a.prevScrollY
    setShowMenu(visible);
  },[a.scrollY])

  useScrollPosition(({ prevPos, currPos }) => {
    setCurrentPositon(currPos.y)
    if (currPos.y>-1){
      // setShowMenu(true)
      return
    }
    const visible = currPos.y > prevPos.y;
    // setShowMenu(visible);
  }, []);

  // submenu
  const submenuref = useRef<HTMLLIElement>(null)
  const [openMenu,setOpenMenu] = useState<boolean>(false)
  useEffect(() => {
    const checkIfClickedOutside = (e:any) => {
      if (openMenu && submenuref.current && !submenuref.current.contains(e.target)) {
        setOpenMenu(false);
        // document.body.style.overflowY = ""
        // document.body.style.touchAction = ""
      }
    };
    document.addEventListener("mousedown", checkIfClickedOutside);
    return () => {
      document.removeEventListener("mousedown", checkIfClickedOutside);
    };
  }, [openMenu]); 

  const windowSize = useWindowDimensions()
  const [width,setWidth] = useState<boolean>()
  function handle(event:any) {
    event.preventDefault();
}
  const setOpenMenuHandler = (e:React.MouseEvent<HTMLLIElement> | undefined) => {
    e?.stopPropagation()
    // console.log("aaaae2")
    openMenu==true?setOpenMenu(false):setOpenMenu(true)
    if (windowSize.width < 768){
      openMenu==false?document.body.style.overflow = "hidden":document.body.style.overflow = ""
      openMenu==false?document.body.style.height = "100%":document.body.style.overflow = ""
      // openMenu==false?document.body.style.touchAction = "none":document.body.style.touchAction = ""
    }else{
    }
  }

  const w = useWindowDimensions()
  const handleStyle = () => {
    if(Props.locationNumber===undefined){
    return {}
    }else if(Props.locationNumber===1){
    return {display:"none"}
    // return {backgroundColor:"#1a252f"}
    }
  }
  const handleStyle2 = () => {
    if(Props.locationNumber===undefined){
    return {}
    }else if(Props.locationNumber===1){
      if(w.width > 768&&currentPostion>-1){
        return {width:"100%",backgroundColor: "rgb(246 246 249 / 30%)"}
      }else if(w.width > 768){
        return {width:"100%",backgroundColor:"#f6f6f9"}
      }else{
        return {width:"100%",backgroundColor: "transparent"}
      }
    }
}
const router = useRouter()
const handleClick = (url:string) => {
  router.push(url)
  dispatch(SubMenuAction(true))
}
// locate
const { t } = useLocale()
const routerHome = () => {
  router.push("/")
}
// console.log(windowSize)
    // useEffect(()=>{
    //   const touchHandler = (event: any) => {
    //     // if(windowSize.width >= 768){
    //     //   return
    //     // }else{
    //       event.preventDefault();
    //     // }
    //   };
    //   document.addEventListener('touchmove', touchHandler, {
    //     passive: false
    //   });
    //   return () => {
    //     document.removeEventListener("touchmove", touchHandler);
    //   };
    // },[windowSize.width])
    const {data,error} = useExecLeft()
    console.log(data)
  return(
    <>
      <div className = "HeaderV1">
        <div className = "HeaderMain"
          style={handleStyle()}
        >
          <div className = "HeaderMainLeft">
            <div className = "HeaderMainLeftTitle">
              <div className = "LOGO">
                <div className=""
                style={{
                  marginRight:"10px"
                }}
                >
                <img src={url('/topimage.png')} alt="Sample image"></img>
                </div>
                <div className = "LogoG">
                  {t.Logo.G}
                </div>
                <div className = "LogoF">
                  {t.Logo.F}
                </div>
                {/* <div className = "LogoHeart">
                  <BsFillSuitHeartFill/>
                </div> */}
              </div>
            </div>
          </div>
        </div>
        <div className = {`HeaderNavi`}  
        style={{position:"relative"}}
        >
          {Props.locationNumber!=1&&(
          <div className = "HeaderMainLeftTitle">
              <div className = "LOGO">
                <div className=""
                onClick={routerHome}
                style={{
                  marginRight:"10px"
                }}
                >
                <img src={url('/topimage.png')} alt="Sample image"></img>
                </div>
                {/* <div className = "LogoG">
                  {t.Logo.G}
                </div>
                <div className = "LogoF">
                  {t.Logo.F}
                </div> */}
                {/* <div className = "LogoHeart">
                  <BsFillSuitHeartFill/>
                </div> */}
            </div>
          </div>
          )}
          <ul className={showMenu?"":"activeScroll"} 
          style={
            handleStyle2()
          }
          >
            <li className = "nomalNavi"><Link href="/"><a><div><div><AiOutlineHome/> {t.Headers.TOP}</div><div className = {"home"}>{t.SubHeader.TOP}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/search"><a><div><div><HiOutlineSearchCircle/> {t.Headers.SEARCH}</div><div className = {"home"}>{t.SubHeader.SEARCH}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/articles"><a><div><div><RiArticleLine/> {t.Headers.ARTICLE}</div><div className = {"home"}>{t.SubHeader.ARTICLE}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/reviews"><a><div><div><AiOutlineComment/> {t.Headers.REVIEWS}</div><div className = {"home"}>{t.SubHeader.REVIEWS}</div></div></a></Link></li>
            <li className = "nomalNavi"><Link href="/threads"><a><div><div><AiOutlineComment/> {t.Headers.THREAD}</div><div className = {"home"}>{t.SubHeader.THREAD}</div></div></a></Link></li>
            
            <li className="headerUserSighIn nomalNavi"
            style={{
              display:"block"
            }}
            ><UserCertification/></li>
            <li 
              className = "subMenu"
              ref={submenuref}
              onClick={setOpenMenuHandler}
            > 
              <IoMdMenu    
              />
              {windowSize.width >= 768&&openMenu==true&&(
              <div className = "subMenuList"
              >
                <div className="">
                  <div onClick={()=>handleClick("/#weekly-ranking")}><a>{t.SubMenu.QUESTIONNAIRE}</a></div>
                  <div onClick={()=>handleClick("/#this-season")}><a>{t.SubMenu.THISSEASON}</a></div>
                  <div onClick={()=>handleClick("/#last-season")}><a>{t.SubMenu.LASTSEASON}</a></div>
                  <div onClick={()=>handleClick("/#next-season")}><a>{t.SubMenu.NEXTSEASON}</a></div>
                  <div onClick={()=>handleClick("/#movies")}><a>{t.SubMenu.MOVIE}</a></div>
                  <div onClick={()=>handleClick("/#news")}><a>{t.SubMenu.NEWS}</a></div>
                  <div onClick={()=>handleClick("/#toptens")}><a>{t.SubMenu.TOP10}</a></div>
                  <div className = "Top100SubMenu"><Link href ="/top100"><a>{t.SubMenu.TOP100}</a></Link></div>
                  <div className = "Top100SubMenu"><Link href ="/tier"><a>{t.SubMenu.TIER}</a></Link></div>
                  <div className = "Top100SubMenu"><Link href ="/weekly"><a>{t.SubMenu.LASTQUESTIONNAIRE}</a></Link></div>
                </div>
              </div>
              )}

                {windowSize.width < 768&&openMenu==true&&(   
                  <>
                  <div className = "subMenuList768"
                  >
                    <div className="subMenuList7681">Main Menu</div>
                  
                    <div className="">
                      <div className = "subMenuList7689"><Link href="/"><a><div className="headersub768"><div className = "home2"><AiOutlineHome/> {t.Headers.TOP}</div><div className = {"home23"}>{t.SubHeader.TOP}</div></div></a></Link>
                        {/* <div className="">Home Menu</div> */}
                        <div className="HederV1SubmenuSub">
                          <div onClick={()=>handleClick("/#weekly-ranking")}><a>{t.SubMenu.QUESTIONNAIRE}</a></div>
                          <div onClick={()=>handleClick("/#this-season")}><a>{t.SubMenu.THISSEASON}</a></div>
                          <div onClick={()=>handleClick("/#last-season")}><a>{t.SubMenu.LASTSEASON}</a></div>
                          <div onClick={()=>handleClick("/#next-season")}><a>{t.SubMenu.NEXTSEASON}</a></div>
                          <div onClick={()=>handleClick("/#movies")}><a>{t.SubMenu.MOVIE}</a></div>
                          <div onClick={()=>handleClick("/#news")}><a>{t.SubMenu.NEWS}</a></div>
                          <div onClick={()=>handleClick("/#toptens")}><a>{t.SubMenu.TOP10}</a></div>
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/search"><a><div className="headersub768"><div className = "home2"><HiOutlineSearchCircle/> {t.Headers.SEARCH}</div><div className = {"home23"}>{t.SubHeader.SEARCH}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          {data.styles.map((item: any) => (
                            <LeftStyle name={item.name} id={item.id} count={item.count} key={item.id}/>
                          ))}
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/articles"><a><div className="headersub768"><div className = "home2"><RiArticleLine/> {t.Headers.ARTICLE}</div><div className = {"home23"}>{t.SubHeader.ARTICLE}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          <LeftsArticles/>
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/reviews"><a><div className="headersub768"><div className = "home2"><AiOutlineComment/> {t.Headers.REVIEWS}</div><div className = {"home23"}>{t.SubHeader.REVIEWS}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          <LeftsReviews/>
                        </div>
                      </div>
                      <div className = "subMenuList7689"><Link href="/threads"><a><div className="headersub768"><div className = "home2"><AiOutlineComment/> {t.Headers.THREAD}</div><div className = {"home23"}>{t.SubHeader.THREAD}</div></div></a></Link>
                        <div className="HederV1SubmenuSub">
                          <LeftsThreads/>
                        </div>
                      </div>
                      <div className = "Top100SubMenu subMenuList7689"><Link href ="/top100"><a><div className="headersub768"><div className = "home2"><AiOutlineHome/>{t.SubMenu.TOP100}</div><div className = {"home23"}></div></div></a></Link></div>
                      <div className = "Top100SubMenu subMenuList7689"><Link href ="/tier"><a><div className="headersub768"><div className = "home2"><AiOutlineHome/>{t.SubMenu.TIER}</div><div className = {"home23"}></div></div></a></Link></div>
                      <div className = "Top100SubMenu subMenuList7689"><Link href ="/weekly"><a><div className="headersub768"><div className = "home2"><AiOutlineComment/> {t.SubMenu.QUESTIONNAIRE}</div><div className = {"home23"}></div></div></a></Link></div>
                    </div>
                  </div>
                  
                  </>  
                )}
            </li>
          </ul>
        </div>
      </div>
    </>
  )
}