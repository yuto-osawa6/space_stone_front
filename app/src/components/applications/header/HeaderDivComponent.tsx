import { ReactNode } from "react"

type Props = {
  children:ReactNode
}

export const HeaderDivComponent:React.FC<Props> = (Props) => {

  return(
    <>
    <div className = "header">
      <div className = "header__box">
        {/* <Header/> */}
        {Props.children}
      </div>
    </div>
    </>
  )
}