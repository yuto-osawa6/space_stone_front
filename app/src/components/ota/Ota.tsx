import { product } from "@/interfaces/product"
import {  products_reds } from "@/lib/api/products_red"
import { useEffect, useState } from "react"


// export const Ota2:React.FC = function Ota2Func(){
//   const [products,setProducts] = useState<product[]>()
//   const [message,setMessage] = useState<string>("")
//   const [count,setCount] = useState<number>(0)

//   useEffect(()=>{
//     const handleOta = async()=>{
//       const res = await fetch('http://localhost:3001/api/v1/products/red')
//       const data = await res.json()
//       setMessage(data.message)
//     }
//     handleOta()
//   },[])
  
//   const handleClick = async() => {
//     const res = await fetch('http://localhost:3001/api/v1/products/red')
//     const data = await res.json()
//     setMessage(data)
//     setCount(1)
//   }

//   return(
//     <>
//     <input
//         id="search"
//         type="text"
//       />
//       <button
//       onClick={handleClick}
//       >
//       {count==1&&(
//         <div className="">aaaaaaaaaaaiu</div>
//       )}
//       </button>
//       {message.length>0?
//       <div className="">
//         Signed in as
//       </div>
//       :
//         <div className="">nocontent</div>
//       }
//     <div className="">Welcome to Next.js!</div>
//       <div>aaaaaa</div>
//     </>
//   )
// }