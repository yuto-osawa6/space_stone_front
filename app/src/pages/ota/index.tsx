import { ShareMain } from "@/components/share/main/ShareMain"
import { productShow } from "@/interfaces/product"
import { execChangeGrid } from "@/lib/api/main"
// import { getCurrentUser } from "@/lib/api/users/sign"
import { useUser } from "@/lib/data/user/useUser"
import { GetServerSideProps, GetStaticProps } from "next"
import Link from "next/link"
import TitleIndex from "@/pages/title/[pid]"
import { ReactNode, useEffect } from "react"
import useSWR from "swr"
import { NextSeo } from 'next-seo';


export const getStaticProps: GetStaticProps = async(context) => {
  // const { pid } = context.query
  const params = {
    active:"1",
    last:"2"
  }
  const query_params = new URLSearchParams(params); 
  const [productShowRes] = await Promise.all([
    fetch(`${process.env.API_PATH_V1}/products/${18}/seo`), 
    // fetch(`${process.env.API_PATH_V1}/mainblocks/mains/pickup?`+ query_params),
    // fetch(`${process.env.API_PATH_V1}/mainblocks/mains/update_tier_list?`+ new URLSearchParams(tierParams))
    // fetch(`${process.env.ApiPathV1}/mainblocks/mains/update_tier_list?`+)
  ]);
  const [data] = await Promise.all([
    productShowRes.json()
  ]);
  return { 
    props: { 
      data
    } 
  };
}

type Props = {
  // getLayout: (page: any) => JSX.Element
  children:ReactNode
  data?:productShow
}

const Ota: React.FC<Props>& { getLayout: (page: any) => JSX.Element } = (Props) => {
  // const {data,error} = getCurrentUser()
  // const { data, error } = useSWR('/session_user')
  const { userSwr, error } = useUser()
  // console.log(data)
  // const changeGridexec = async() => {
  //   // console.log(grid)
  //   const res = await execChangeGrid("01")
  //   if (res.status === 200) {
  //     // dispatch(GridAction(res.data.grid))
  //   }
  // }

  // useEffect(()=>{
  //   console.log(window)
  //   // if(typeof window !== 'undefined')return
  //   changeGridexec()
  // },[])
  console.log(Props)
  return(
    <>
      <NextSeo
       title={`${Props.data?.products.title} - SpaceTone`}
      //  description={Props.data.products.}
      >
      </NextSeo>
      {/* <ShareMain> */}
        {/* aaaaaaaaajklkaaaaafl */}
        {/* {userSwr.user?.nickname} */}
      {/* </ShareMain> */}
      {/* {Props.data!=undefined&&(
      <TitleIndex
      data={Props.data}
      >

      </TitleIndex>
      )} */}
      <Link href="/ota/po">
          <a>Home</a>
      </Link>
      <div className="">
        {Props.data?.products.imageUrl}
      </div>
      aaaaaaallllllllllllllaaa
      <div className="">
        {Props.children}
      </div>
      bbbbbbb

      <Link href={`/title/${Props.data?.products.id}`}>
          <a>Home</a>
      </Link>

    </>
  )
}

export default Ota


Ota.getLayout = function getLayout(page) {
  return (
    <ShareMain>
      {page}
    </ShareMain>
  )
}