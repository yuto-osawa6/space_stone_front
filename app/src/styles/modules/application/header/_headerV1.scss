.HeaderV1{
  // height: 59.66px;
  @include mq(md){
  display: grid;
  grid-template-columns: 200px 1fr;
  }
}
.header{
  // top: 0;
  // position: sticky;
  // z-index: 200;
  // top: 0;
  // position: sticky;
  // z-index: 200;
  

}
.activeScroll{
  transform: translateY(-100px);
}

.HeaderGrid0{
  // background-color:$color-v1-g0;
  background-color:$color-v1-h0;
}
.HeaderGrid1{
  // background-color:$color-v1-g1;
  background-color:$color-v1-h1;
}
.HeaderGrid2{
  // background-color:$color-v1-g2;
  background-color:$color-v1-h2;
}
.HeaderGrid3{
  // background-color:$color-v1-g3;
  background-color:$color-v1-h3;
}
.HeaderGrid4{
  // background-color:$color-v1-g4;
  background-color:$color-v1-h4;
}
.HeaderGrid5{
  // background-color:$color-v1-g5;
  background-color:$color-v1-h5;
}
.HeaderGrid6{
  // background-color:$color-v1-g10;
  background-color:$color-v1-h10;
}
.HeaderGrid7{
  // background-color:$color-v1-g7;
  background-color:$color-v1-h7;
}
.HeaderGrid8{
  // background-color:$color-v1-g8;
  background-color:$color-v1-h8;
}
.HeaderGrid9{
  // background-color:$color-v1-g9;
  background-color:$color-v1-h9;
}

.HeaderGrid12{
  // background-color: white;
  background-color: #1a252f;
}

.HeaderGrid11{
  // background-color: $color-v1-d;
  background-color: #1a252f;
}

// $color-v1-g0:#f7f48a;
// $color-v1-g1:#99f9c8;
// $color-v1-g2:#f7c280;
// $color-v1-g3:#ff9292;
// $color-v1-g4:#f5b7ed;
// $color-v1-g5:#f0aaf9;
// $color-v1-g6:#9bc0f7;
// $color-v1-g7:#ccf9b0;
// $color-v1-g8:#b5b8ff;
// $color-v1-g9:#bdffb5;
// $color-v1-g10:#ff9eb9;

.LOGO{
  display: flex;
  width: fit-content;
  position: relative;
  .LogoG{
    font-size: 1.8rem;
    color: #ff3073;
    z-index: 100;
  }
  .LogoF{
    font-size: 1.8rem;
    color: white;
    /* position: absolute; */
    top: 0;
    right: px;
    z-index: 10;
    position: relative;
  }
  .LogoHeart{
    position: relative;
    right: 20px;
    /* background-color: aqua; */
    z-index: 9;
    color: aqua;
    svg{
      width: 30px;
      height: 30px;
      background-color: transparent;
      color: #ff3073;
      /* position: absolute; */
      top: 0;
      color: white;
      color: #ff3073;
    }
  }
}
.HeaderMain{
  display: none;
  @include mq(md){
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #1a252f;
  }
}
.HeaderMainLeft{
  padding: 10px 20px 0px 20px;
  .HeaderMainLeftTitle{
    margin-right: -20px;
  }
  // background-color: #1b548f;
}
.HeaderNavi{
  position: relative;
  // background-color: #edf1f5;
  background-color: #ffffff;
  // font-size: 0.9rem;
  ul{
    // background-color: #edf1f5;
    padding-top:10px;
    padding-bottom:10px;
    max-width: 1300px;
    background-color: #1a252f;
    transition: 1s;
    position: fixed;
    height: 59.66px;
    z-index: 300;
    display: flex;
    justify-content: center;
    gap: 20px;
    align-items: center;
    width: 100%;
    font-size: 0.9rem;;
    @include mq(md){
      font-size: 1.0rem;
      background-color: #ffffff;
      width: calc(100% - 200px);
    }

    li{
      color: white;
      // background-color: #167eff;
      background-color: #1a252f;
      // padding: 5px 15px 5px 15px;
      // border: 1px solid grey;
      border-radius: 30px;
      box-shadow: $color-v1-shadow-a;
      a{
      padding: 5px 15px 5px 15px;
      cursor: pointer;
      display: block;
      }
    }
    .headerUserSighIn{
      background-color: $color-v1-d;
      padding: 0px;
      .user_box_false{
        .user_mypage{
        display: flex;
        gap: 5px;
        align-items: center;
        padding: 5px 10px 5px 10px;
        cursor: pointer;
        }
        .userNavigationDummy{
          position: relative;
          .userNavigationDummyAbusolute{
            font-size: 0.8rem;
            // doneyet 確認 文字の長さではみ出る問題
            overflow-x: hidden;
            max-width: 200%;
            position: absolute;
            min-width: 100%;
            background-color: #1a252f;;
            top: 20px;
            padding: 5px;
            border-radius: 5px;
            display: flex;
            gap: 5px;
            flex-flow: column;
            .UserNavigateDummyAbusoluteTop{
              border-bottom: 1px solid;
              display: flex;
              gap: 10px;
              align-items: center;
              padding: 5px;
     
              .UserNavigateDummyAbusoluteTopImg{
                display: flex;
                justify-content: center;
                img{
                  // display: flex;
                  // justify-content: center;
                  width: 30px;
                  border-radius: 30px;
                  height: 30px;
                }
              }
              .UserNavigateDummyAbusoluteTopName{
                font-size: 0.9rem;
                text-align: center;
                margin-top: 10px;
                padding-bottom: 5px;
                // border-bottom: 1px solid;
              }
            }
            
            .user_logout{
              padding: 5px 15px 5px 15px;
              box-shadow: $color-v1-shadow-a;
              display: flex;
              align-items: center;
              gap: 5px;
              padding: 5px;
              &:hover{
                cursor: pointer;
                color: #ff3073;
              }
            }
          }
        }
      }
      .user_box_true{
        padding: 5px 10px 5px 10px;
        .user_sign_in{
          .model_btn{
          display: flex;
          gap: 5px;
          align-items: center;
          }
          // padding: 5px;
        }
      }
    }
  }
}

// userSettingModal

.settingModalOpenClass{
  max-width: 700px;
  background-color: white;
  border-radius: 5px;
  top: 50%;
  position: absolute;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 20px;
  min-width: 500px;
  .settingModalOpenClassTitle{
    font-size: 1.5rem;
  }
  .settingModalOpenClassMain{
    .settingModalOpenClassNickname{
      margin-top: 10px;
      width: 100%;
    }
    .settingModalOpenClassImage{
      font-size: 1.2rem;
      margin-top: 10px;
    }
    .settingModalOpenClassImageExplain{
      color: #4c5966;
      margin: 10px 10px;
      font-size: 0.9rem;
      a{
        color: #1a76d2;
      }
    }
  }
}

// sub
.subMenu{
  padding: 16px;
  position: relative;
  svg{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
  .subMenuList{
    position: absolute;
    right: 0px;
    top: 32px;
    background-color: #1a252f;
    width: fit-content;
    margin-top: 25px;
    padding: 10px;
    border-radius: 5px;
    display: flex;
    gap: 10px;
    font-size: 0.8rem;
    flex-flow: column;
    div{
      cursor: pointer;
      padding: 3px;
      white-space: nowrap;
      &:hover{
        color: #1a76d2;;
      }
    }
  }
}