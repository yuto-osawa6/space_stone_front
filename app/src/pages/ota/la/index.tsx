import { ShareMain } from "@/components/share/main/ShareMain"
import { productShow } from "@/interfaces/product"
import { execChangeGrid } from "@/lib/api/main"
// import { useGetCurrentUser } from "@/lib/api/users/sign"
import { useUser } from "@/lib/data/user/useUser"
import { GetServerSideProps, GetStaticProps } from "next"
import Link from "next/link"
import TitleIndex from "@/pages/title/[pid]"
import { useEffect } from "react"
import useSWR from "swr"
import { Ota2 } from "@/components/ota/Ota"
import Ota from ".."



// export const  getStaticProps: GetStaticProps = async(context) => {
//   // const { pid } = context.query
//   // const params = {
//   //   active:"1",
//   //   last:"2"
//   // }
//   // const query_params = new URLSearchParams(params); 
//   // const [productShowRes] = await Promise.all([
//   //   fetch(`${process.env.API_PATH_V1}/products/red`), 
//   // ]);
//   const res = await fetch(`${process.env.API_PATH_V1}/products/red`)
//   const data = await res.json()

//   // if (productShowRes.status !== 200) {
//     if (res.status !== 200) {
//       return {
//         notFound: true
//       };
//     }
    
//   // const [data] = await Promise.all([
//   //   productShowRes.json()
//   // ]);
  
//   return { 
//     props: { 
//       data
//     } 
//   };
// }

type Props = {
  // getLayout: (page: any) => JSX.Element
  // children:ReactNode
  // data?:productShow
  // data:{
  //   message:string
  // }
}


const Home2: React.FC<Props>& { getLayout: (page: any) => JSX.Element } = (Props) => {
  // console.log(Props.data)
  // const {data,error} = getCurrentUser()
  // const { data, error } = useSWR('/session_user')
  // const { userSwr, error } = useUser()
  // console.log(data)
  // const changeGridexec = async() => {
  //   // console.log(grid)
  //   const res = await execChangeGrid("01")
  //   if (res.status === 200) {
  //     // dispatch(GridAction(res.data.grid))
  //   }
  // }

  // useEffect(()=>{
  //   console.log(window)
  //   // if(typeof window !== 'undefined')return
  //   changeGridexec()
  // },[])


  return(
    <>
     {/* Ota2>
      aaa aaa
      <Ota2></
      {Props.data.message.length>0?
      <div className="">
        tests
      </div>
      :
        <div className="">nocontent</div>
      } */}
    </>
  )
}

export default Home2


Home2.getLayout = function getLayout(page) {
  return (
    // <ShareMain>
    <div className="">
      {page}
    </div>
  )
}
