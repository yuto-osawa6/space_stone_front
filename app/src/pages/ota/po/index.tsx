import { ShareMain } from "@/components/share/main/ShareMain"
import { productShow } from "@/interfaces/product"
import { execChangeGrid } from "@/lib/api/main"
// import { getCurrentUser } from "@/lib/api/users/sign"
import { useUser } from "@/lib/data/user/useUser"
import { GetServerSideProps, GetStaticProps } from "next"
import Link from "next/link"
import TitleIndex from "@/pages/title/[pid]"
import { useEffect } from "react"
import useSWR from "swr"
import { Ota2 } from "@/components/ota/Ota"
import Ota from ".."


// type Props = {
//   // getLayout: (page: any) => JSX.Element
// }

// export const  getStaticProps: GetStaticProps = async(context) => {
//   // const { pid } = context.query
//   const params = {
//     active:"1",
//     last:"2"
//   }
//   const query_params = new URLSearchParams(params); 
//   const [productShowRes] = await Promise.all([
//     fetch(`${process.env.API_PATH_V1}/products/${1}`), 
//     // fetch(`${process.env.API_PATH_V1}/mainblocks/mains/pickup?`+ query_params),
//     // fetch(`${process.env.API_PATH_V1}/mainblocks/mains/update_tier_list?`+ new URLSearchParams(tierParams))
//     // fetch(`${process.env.ApiPathV1}/mainblocks/mains/update_tier_list?`+)
//   ]);
//   const [data] = await Promise.all([
//     productShowRes.json()
//   ]);
//   return { 
//     props: { 
//       data
//     } 
//   };
// }

type Props = {
  // getLayout: (page: any) => JSX.Element
  // children:ReactNode
  // data?:productShow
}


const Otare: React.FC<Props>& { getLayout: (page: any) => JSX.Element } = (Props) => {
  // const {data,error} = getCurrentUser()
  // const { data, error } = useSWR('/session_user')
  const { userSwr, error } = useUser()
  // console.log(data)
  // const changeGridexec = async() => {
  //   // console.log(grid)
  //   const res = await execChangeGrid("01")
  //   if (res.status === 200) {
  //     // dispatch(GridAction(res.data.grid))
  //   }
  // }

  // useEffect(()=>{
  //   console.log(window)
  //   // if(typeof window !== 'undefined')return
  //   changeGridexec()
  // },[])


  return(
    <>
      aaa
      {/* <Ota
      data={Props.data}
      >
      </Ota> */}
      <Ota2></Ota2>
      aaa
      {/* <Link href="/ota">
          <a>pome</a>
      </Link>
      <Ota
        data={Props.data}
      >
        kkkkkkkkkkkkkkkkkkkkkkkkkkk
      </Ota>
      <Link href="/ota/po/1">
          <a>pome</a>
      </Link> */}
    </>
  )
}

export default Otare


Otare.getLayout = function getLayout(page) {
  return (
    <ShareMain>
      {page}
    </ShareMain>
  )
}

// const Otare: React.FC<Props> = () => {
//   return(
//     <><>
//   )
// }