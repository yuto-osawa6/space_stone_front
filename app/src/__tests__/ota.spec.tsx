import { render, screen } from '@testing-library/react'
import Home from '../pages/index'
import '@testing-library/jest-dom'
import { Header } from '@/components/applications/header/Header';
import { Ota } from '@/components/ota/Ota';
// import { LeftsArticles } from 'components/applications/left/leftMenus/LeftArticles'

describe("Sampleコンポーネント", () => {
  // ダミーのpropsを定義
  // let dummyProps: ;
  // beforeEach(() => {
  //   dummyProps = {
  //     userId: 1,
  //     id: 1,
  //     title: 'dummy title 1',
  //     body: 'dummy body 1',
  //   };
  // });
  test("should first", () => {
    render(<Ota></Ota>)
    // // render(<Header
    // //   locationNumber={1}
    // //   />)
    // render(
    //   // <LeftsArticles></LeftsArticles>
    // )
    // const { getByText } = render(<Sample />);
    // expect(getByText("Nextjs+Jestのサンプルサプリ")).toBeTruthy();
    expect(1+1).toBe(2);
  });
});