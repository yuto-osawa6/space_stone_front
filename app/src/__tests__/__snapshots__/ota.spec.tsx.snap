// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App false is falsy 1`] = `
<div>
  <input
    id="search"
    type="text"
  />
  <button />
  <div
    class=""
  >
    Signed in as
  </div>
  <div
    class=""
  >
    Welcome to Next.js!
  </div>
  <div>
    aaaaaa
  </div>
</div>
`;
